src/app/gui/map/mapservice.js

```diff
@@ -1691,64 +1709,90 @@ proto._resetView = function() {
-      this.zoomToGeometry(geometry);
-    });
-  const initextent = map_extent ? map_extent.split(',').map(coordinate => 1*coordinate) : project.state.initextent;
-  const projection = this.getProjection();
-  const extent = project.state.extent;
-  const maxxRes = ol.extent.getWidth(extent) / width;
-  const maxyRes = ol.extent.getHeight(extent) / height;
-  const maxResolution = Math.max(maxxRes,maxyRes);
-  const initxRes = ol.extent.getWidth(initextent) / width;
-  const inityRes = ol.extent.getHeight(initextent) / height;
-  const resolution = Math.max(initxRes,inityRes);
-  const center = ol.extent.getCenter(initextent);
+/**
+ * @param project
+ * @param width
+ * @param height
+ * @param { Array } map_extent since 3.10.0: in case of true, use url parameter to set view options
+ * @return {{extent: *, maxResolution: number, center: *, projection: *, resolution: number}}
+ * @private
+ */
+proto._calculateViewOptions = function({
+  project,
+  width,
+  height,
+  map_extent,
+} = {}) {
+  const initextent = map_extent ? map_extent.split(',').map(coord => 1 * coord) : project.state.initextent;
+  const extent     = project.state.extent;
   return {
-    projection,
-    center,
     extent,
-    maxResolution,
-    resolution
+    projection:    this.getProjection(),
+    center:        ol.extent.getCenter(initextent),
+    maxResolution: Math.max(ol.extent.getWidth(extent) / width,     ol.extent.getHeight(extent) / height),     // max(xRes, yRes)
+    resolution:    Math.max(ol.extent.getWidth(initextent) / width, ol.extent.getHeight(initextent) / height), // max(xInitRes, yInitRes)
   }
 };
 
 // set view based on project config
 proto._setupViewer = function(width, height) {
+  const search           = new URLSearchParams(location.search); // search params
+  const showmarker       = 1 * (search.get('showmarker') || 0);  /** @since 3.10.0 0 or 1. Show marker on map center*/
+  const iframetype       = search.get('iframetype');             /** @since 3.10.0 type of iframe: map (only map, no control)*/
+  const map_extent       = search.get('map_extent');
+  const zoom_to_fid      = search.get('zoom_to_fid');
+  const zoom_to_features = search.get('ztf');                    // zoom to features
+  const coords           = {
+      lat: 1 * search.get('lat'),
+      lon: 1 * search.get('lon'),
+      x: 1 * search.get('x'),
+      y: 1 * search.get('y'),
+    };
+
+  (new Promise((resolve) => {
+    setTimeout(async () => {
+      let geom;
+      if (zoom_to_fid) {
+        await this.zoomToFid(zoom_to_fid);
+      } else if (zoom_to_features) {
+        await this.handleZoomToFeaturesUrlParameter({ zoom_to_features });
+      } else if (coords.lat && coords.lon && !Number.isNaN(coords.lat) && !Number.isNaN(coords.lon)) {
+        geom = new ol.geom.Point(ol.proj.transform([coords.lon, coords.lat], 'EPSG:4326', this.getEpsg()));
+      } else if (coords.x && coords.y && !Number.isNaN(coords.x) && !Number.isNaN(coords.y)) {
```

Comment:
@volterra79 Shouldn't that be an acceptable value?

```
coords.x = 0
coords.x = 0
```

Reply:
@Raruto right. Need to check if coord.x && coord.y are not undefined or null. 0, 0 it is acceptable